pgmpy/factors/continuous/ContinuousFactor.py

```diff
@@ -0,0 +1,393 @@
+        >>> std_normal.scope()
+        ['x1']
+        >>> std_normal.assignment(1)
+
+        """
+        if not isinstance(variables, (list, tuple, np.ndarray)):
+            raise TypeError("variables: Expected type list or array-like, "
+                             "got type {var_type}".format(var_type=type(variables)))
+
+        if not all([var in self.variables for var in variables]):
+            raise ValueError("Variable not in scope.")
+
+        phi = self if inplace else self.copy()
+
+        for var in variables:
+            if var not in phi.variables:
+                raise ValueError("{var} not in scope.".format(var=var))
+
+        all_var = [var for var in self.variables]
+        var_to_keep = [var for var in self.variables if var not in variables]
+        reordered_var_index = [all_var.index(var) for var in variables + var_to_keep]
+        pdf = phi.pdf
+
+        # The arguments need to be reordered because integrate.nquad integrates the first n-arguments
+        # of the function passed.
+
+        def reordered_pdf(*args):
+            # ordered_args restores the original order as it was in self.variables
+            ordered_args = [args[reordered_var_index.index(index_id)] for index_id in range(len(all_var))]
+            return pdf(*ordered_args)
+
+        def marginalized_pdf(*args):
+            return integrate.nquad(reordered_pdf, [[-np.inf, np.inf] for i in range(len(variables))], args=args)[0]
+
+        phi.pdf = marginalized_pdf
+        phi.variables = var_to_keep
+
+        if not inplace:
+            return phi
+
+    def normalize(self, inplace=True):
+        """
+        Normalizes the pdf of the continuous factor so that it integrates to
+        1 over all the variables.
+
+        Parameters
+        ----------
+        inplace: boolean
+            If inplace=True it will modify the factor itself, else would return
+            a new factor.
+
+        Returns
+        -------
+        ContinuousFactor or None:
+             if inplace=True (default) returns None
+             if inplace=False returns a new ContinuousFactor instance.
+
+        Examples
+        --------
+        >>> from pgmpy.factors import ContinuousFactor
+        >>> from scipy.stats import multivariate_normal
+        >>> std_normal_pdf = lambda x: 2 * multivariate_normal.pdf(x, [0, 0], [[1, 0], [0, 1]])
+        >>> std_normal = ContinuousFactor(['x1', 'x2'], std_normal_pdf)
+        >>> std_normal.assignment(1, 1)
+        0.117099663049
+        >>> std_normal.normalize()
+        >>> std_normal.assignment(1, 1)
+        0.0585498315243
+
+        """
+        phi = self if inplace else self.copy()
+        pdf = self.pdf
+
+        pdf_mod = integrate.nquad(pdf, [[-np.inf, np.inf] for var in self.variables])[0]
+
+        phi.pdf = lambda *args: pdf(*args) / pdf_mod
+
+        if not inplace:
+            return phi
+
+    def operate(self, other, operation, inplace=True):
```

Comment:
@yashu-seth Add product and divide methods. Rename this to `_operate` to make it an internal method.